### ISSUE TITLE
option to ignore blank lines with `uniq`

### ISSUE THREAD

**Issue by ShinyZero0** [nushell/nushell#9122]
### Related problem

I was modifying my `.gitignore` with nu to add some paths and then applied `uniq` and all my blank lines formatting gone, which is not cool

### Describe the solution you'd like

Add an option for `uniq` to ignore empty lines/nulls or an option to specify which kind of values should be ignored like `uniq --ignore ""` or `uniq --ignore '^\s*$'`

**Comment by amtoine** [comment]
@ShinyZero0 
is the problem that, for instance
```
[
    2,
    "",
    "",
    1,
    4,
    "",
    3
    "",
    "",
] | uniq
```
will give
```
╭───┬───╮
│ 0 │ 2 │
│ 1 │   │
│ 2 │ 1 │
│ 3 │ 4 │
│ 4 │ 3 │
╰───┴───╯
```
?

what would be the right output when ignoring empty lines?

**Comment by kubouch** [comment]
You should be able to easily filter them out, e.g., `open .gitignore | lines | where not ($it | is-empty) | uniq`. I'd rather not complicate our commands further if the same functionality can be achieved which a combination of existing commands.

**Comment by ShinyZero0** [comment]
> You should be able to easily filter them out, e.g., `open .gitignore | lines | where not ($it | is-empty) | uniq`. I'd rather not complicate our commands further if the same functionality can be achieved which a combination of existing commands.

No, it's no what i want to achieve. I want to preserve blank lines after `uniq` so that when i save the file it won't break my formatting.

**Comment by ShinyZero0** [comment]
> @ShinyZero0 is the problem that, for instance
> 
> ```
> [
>     2,
>     "",
>     "",
>     1,
>     4,
>     "",
>     3
>     "",
>     "",
> ] | uniq
> ```
> 
> will give
> 
> ```
> ╭───┬───╮
> │ 0 │ 2 │
> │ 1 │   │
> │ 2 │ 1 │
> │ 3 │ 4 │
> │ 4 │ 3 │
> ╰───┴───╯
> ```
> 
> ?
> 
> what would be the right output when ignoring empty lines?

Sorry, the more fitting word there would be preserve. So that when `uniq` does some loop and checking if value was met before it won't check the value if it matches a predicate e.g. `{ || is-empty }` and just add it to the resulting collection

**Comment by kubouch** [comment]
Can you give an example of input and expected output?

**Comment by ShinyZero0** [comment]
[add.txt](https://github.com/nushell/nushell/files/11415013/add.txt)
[gitignore.txt](https://github.com/nushell/nushell/files/11415015/gitignore.txt)
if you have e.g. these files and you want to merge the new ignore rules into gitignore like
```
open gitignore.txt | lines | prepend (open add.txt | lines) | uniq
```
blank lines will be removed while the result i'd expect is
[result.txt](https://github.com/nushell/nushell/files/11415020/result.txt)
